cloud/services/object_storage_buckets.go

```diff
@@ -114,3 +114,18 @@ func CreateObjectStorageKeys(ctx context.Context, bucketScope *scope.ObjectStora
 
 	return newKeys, nil
 }
+
+func DeleteObjectStorageKeys(ctx context.Context, bucketScope *scope.ObjectStorageBucketScope, logger logr.Logger, keys [2]float64) error {
+
+	for _, keyID := range keys {
+		logger.Info("deleting object storage access key %s", keyID)
+		if err := bucketScope.LinodeClient.DeleteObjectStorageKey(ctx, int(keyID)); err != nil {
+			logger.Info("access key does not exist")
+		} else if err != nil {
```

Comment:
This is a duplicate error check. Did you mean for the check above to eval if the error is a 404?